### ISSUE TITLE
Knex doesn't parse JSON values in connectionURIs causing incompatibility with mysql / node-mysql2 

### ISSUE THREAD

**Issue by petetnt** [knex/knex#4628]
# Environment

Knex version: 0.95.9, 0.21.21
Database + version: MariaDB 10.3, `mysql` package 2.18.1 but also affects node-mysql2
OS: MacOS Monterey

# Bug

## 1. Explain what kind of behaviour you are getting and how you think it should do

I am trying to pass in a connectionURI that contains JSON in the `searchParams`.

```
mariadb://foo:bar@foobar.com/foo?ssl={\"ca\": \"--- BEGIN CERTIFICATE --- ...\"}
```

This might seem funky at first, but it is actually supported by both `mysql` and `node-mysql2`

Here's the `node-mysql2` implementation:

https://github.com/sidorares/node-mysql2/blob/d74558b605162156b813248a024f7559785de6fb/lib/connection_config.js#L234-L254

and here's the `mysql` one:

https://github.com/mysqljs/mysql/blob/3430c513d6b0ca279fb7c79b210c9301e9315658/lib/ConnectionConfig.js#L179-L209

As you can see, the parser first tries to parse the search/query parameters with `JSON.parse` and then fallbacks on just adding the value as string.

`knex`s parser, however, just assigns them as strings:

https://github.com/knex/knex/blob/1744c8c2655a8362260ba987d706ec3473681a78/lib/knex-builder/internal/parse-connection.js#L70-L73

## 2. Error message

The SSL configuration always gets parsed as a string, resulting (for example in `mysql`) in the following:

```
TypeError: Unknown SSL profile '{"ca":"-----BEGIN CERTIFICATE-----
```

## 3. Reduced test code, for example in https://npm.runkit.com/knex or if it needs real
   database connection to MySQL or PostgreSQL, then single file example which initializes
   needed data and demonstrates the problem.

Codesandbox:
https://codesandbox.io/s/dazzling-burnell-7gsyj?file=/src/index.js:0-197

```js
const knex = require("knex")({
  client: "mysql",
  connection:
    'mariadb://foo:bar@foobar.com/foo?ssl={"ca": "--- BEGIN CERTIFICATE --- ..."}'
});

console.log(knex.client.connectionSettings);
```

Result:

```js
{
  database: 'foo',
  host: 'foobar.com',
  user: 'foo',
  password: 'bar',
  ssl: '{"ca": "--- BEGIN CERTIFICATE --- ..."}'
}
```

Expected:

```js
{
  database: 'foo',
  host: 'foobar.com',
  user: 'foo',
  password: 'bar',
  ssl: : {
    ca: "--- BEGIN CERTIFICATE --- ..."
  }
}
```

**Comment by kibertoad** [comment]
Would you be open to send a PR for this?

**Comment by petetnt** [comment]
Hi @kibertoad!

Sent a PR in https://github.com/knex/knex/pull/4629 (first time using GitHub Codespaces to do it 😸 , hopefully it works & tests pass, if not I'll do the changes later locally :))

**Comment by kibertoad** [comment]
Released in 0.95.11
